api/src/pcapi/core/offers/api.py

```diff
@@ -1528,3 +1529,125 @@ def update_offers_description_from_product_description(product_id: int) -> None:
             extra={"product": product_id, "offers": offer_ids, "exc": str(exception)},
         )
         raise exceptions.NotUpdateProductOrOffers(exception)
+
+
+def check_can_move_event_offer(offer: models.Offer) -> list[offerers_models.Venue]:
+    if not offer.isEvent:
+        raise exceptions.OfferIsNotEvent()
+
+    count_past_stocks = (
+        models.Stock.query.with_entities(models.Stock.id)
+        .filter(models.Stock.beginningDatetime < datetime.datetime.utcnow())
+        .count()
+    )
+    if count_past_stocks > 0:
+        raise exceptions.OfferEventInThePast(count_past_stocks)
+
+    count_reimbursed_bookings = (
+        bookings_models.Booking.query.with_entities(bookings_models.Booking.id)
+        .join(bookings_models.Booking.stock)
+        .filter(models.Stock.offerId == offer.id, bookings_models.Booking.isReimbursed)
+        .count()
+    )
+    if count_reimbursed_bookings > 0:
+        raise exceptions.OfferHasReimbursedBookings(count_reimbursed_bookings)
+
+    venues_choices = (
+        offerers_models.Venue.query.filter(
+            offerers_models.Venue.managingOffererId == offer.venue.managingOffererId,
+            offerers_models.Venue.id != offer.venueId,
+        )
+        .join(
+            offerers_models.VenuePricingPointLink,
+            sa.and_(
+                offerers_models.VenuePricingPointLink.venueId == offerers_models.Venue.id,
+                offerers_models.VenuePricingPointLink.timespan.contains(datetime.datetime.utcnow()),
+            ),
+        )
+        .options(
+            sa.orm.load_only(offerers_models.Venue.id, offerers_models.Venue.name),
+            sa.orm.contains_eager(offerers_models.Venue.pricing_point_links).load_only(
+                offerers_models.VenuePricingPointLink.pricingPointId, offerers_models.VenuePricingPointLink.timespan
+            ),
+        )
+        .all()
+    )
+    if not venues_choices:
+        raise exceptions.NoDestinationVenue()
+
+    return venues_choices
+
+
+def move_event_offer(
+    offer: models.Offer, destination_venue: offerers_models.Venue, notify_beneficiary: bool = False
+) -> None:
+    """
+    - offer MUST have been checked by check_can_move_event_offer() before calling this function
+    - when offer has priced bookings, pricing point for destination venue must be the same
```

Comment:
Je suis pas bien sûr de comprendre cette ligne. Enfin j'ai peur que le "must be the same" ne soit pas très clair...